nix/overlay.nix

```diff
@@ -49,15 +49,18 @@ let
       '';
     };
 
+  sources = import ./sources.nix;
+  pkgsCargo = import sources.nixpkgs-cargo {};
```

Comment:
I'm not sure if we (still) do some custom Darwin stuff when instantiating the regular nixpkgs, you might want to compare it with the invocation there. But we can also deal with it in a followup, no need to block this PR.